inference-engine/src/inference_engine/src/ie_network_reader.cpp

```diff
@@ -357,54 +357,66 @@ CNNNetwork convert_to_cnnnetwork(std::shared_ptr<ngraph::Function>& function,
-                auto old_api_type = old_api_map_attr_val.get_type();
-                const auto old_api_transpose_args = old_api_map_attr_val.get_order();
-
-                OPENVINO_ASSERT(!old_api_type.is_dynamic(), "Old API map does not support dynamic type");
-                // if no differences between IR v10 and IR v11, add identity convert which will be optimized out
-                if (old_api_type == ov::element::undefined)
-                    old_api_type = parameter->get_element_type();
-
-                prepost.input(ov::preprocess::InputInfo(i)
-                                  .tensor(InputTensorInfo().set_element_type(old_api_type))
-                                  .preprocess(PreProcessSteps().convert_layout(old_api_transpose_args)));
+                std::vector<uint64_t> order;
+                ngraph::element::Type type;
+                const auto it_type = rtInfo.find(old_api_map_key_type);
+                if (it_type != rtInfo.end()) {
+                    const auto old_api_map_attr = std::dynamic_pointer_cast<ov::OldApiMapElementType>(it_type->second);
+                    OPENVINO_ASSERT(old_api_map_attr != nullptr, "Failed to cast to ov::OldApiMapElementType");
+                    type = old_api_map_attr->get();
+
+                    OPENVINO_ASSERT(!type.is_dynamic(), "Old API map does not support dynamic type");
+                }
+                const auto it_order = rtInfo.find(old_api_map_key_order);
+                if (it_order != rtInfo.end()) {
+                    const auto old_api_map_attr = std::dynamic_pointer_cast<ov::OldApiMapOrder>(it_order->second);
+                    OPENVINO_ASSERT(old_api_map_attr != nullptr, "Failed to cast to ov::OldApiMapOrder");
+                    const auto old_api_map_attr_val = old_api_map_attr->get();
+                    order = old_api_map_attr_val.get_order();
+                }
+                if (it_type != rtInfo.end() && it_order != rtInfo.end())
+                {
+                    prepost.input(ov::preprocess::InputInfo(i)
+                                          .tensor(InputTensorInfo().set_element_type(type))
+                                          .preprocess(PreProcessSteps().convert_layout(order)));
+                    rtInfo.erase(it_type);
+                    rtInfo.erase(it_order);
+                }
+                else if (it_type != rtInfo.end()) {
+                    prepost.input(ov::preprocess::InputInfo(i)
+                                          .tensor(InputTensorInfo().set_element_type(type)));
+                    rtInfo.erase(it_type);
+                }
+                else if (it_order != rtInfo.end()) {
+                    prepost.input(ov::preprocess::InputInfo(i)
+                                          .tensor(InputTensorInfo())
+                                          .preprocess(PreProcessSteps().convert_layout(order)));
+                    rtInfo.erase(it_order);
+                }
```

Comment:
Please, use more convenient syntax from https://github.com/openvinotoolkit/openvino/pull/8629
Now we don't have a lot of if / else

CC @nosovmik

Reply:
Done.